active/0021-transfer-files-over-mqtt.md

```diff
@@ -70,42 +70,68 @@ As an example of existing implementation we can look at AWS IoT Core [which prov
-* Broker MUST validate that provided UUID conforms to [RFC 4122](https://www.rfc-editor.org/rfc/rfc4122)
+* Client MAY generate UUID according to [RFC 4122](https://www.rfc-editor.org/rfc/rfc4122) for each file being transferred and use it as `fileId` in Topic Name
 * Client MAY calculate SHA-256 checksum of the segment it's about to send and send it as part of Topic Name
 * Client MAY calculate SHA-256 checksum of the file it's about to send and include it in the `init` message payload or send is as part of the `fin` message
 * If Client chooses to provide checksum for file segments, whole file, or both, it MUST use [SHA-256](https://www.rfc-editor.org/rfc/rfc6234)
-* If checksum is included in the `init` message payload, the Broker MUST use it to verify integrity of the file after receving the `fin` message for the corresponding file transfer
+* If checksum is included in the `init` message payload, the Broker MUST use it to verify integrity of the file after receiving the `fin` message for the corresponding file transfer
 * If checksum is included in topic name, Broker MUST use it to verify integrity of corresponding data:
   * segment, if it's a segment transfer message
   * whole file, if it's a `fin` message
 * If checksum verification fails, Broker MUST reject the corresponding data
-* Client MUST use $file Topic to transfer files
-* Broker MUST NOT let clients subscribe to $file topics
+* Client MUST use Topic starting with `$file/` to transfer files
+* Broker MUST NOT let clients subscribe to Topics starting with `$file/` topics
 * Segment length can be calculated on the server side by subtracting the length of the [Variable Header](https://docs.oasis-open.org/mqtt/mqtt/v5.0/os/mqtt-v5.0-os.html#_Toc3901025) from the [Remaining Length](https://docs.oasis-open.org/mqtt/mqtt/v5.0/os/mqtt-v5.0-os.html#_Toc3901105) field that is in the [Fixed Header](https://docs.oasis-open.org/mqtt/mqtt/v5.0/os/mqtt-v5.0-os.html#_Toc3901021)
-* Data is transferred in PUBLISH packets in the following order:
-  1. $file/{fileId}/init
-  2. $file/{fileId}/{offset}[/{checksum}]
-  3. $file/{fileId}/{offset}[/{checksum}]
-  4. ...
-  3. $file/{fileId}/[fin[/{checksum}] | abort]
+
+### Protocol flow
+
+Data is transferred in PUBLISH packets in the following order:
+
+1. `$file/{fileId}/init`
+2. `$file/{fileId}/{offset}[/{checksum}]`
+3. `$file/{fileId}/{offset}[/{checksum}]`
+4. ...
+5. `$file/{fileId}/[fin/{fileSize}[/{checksum}] | abort]`
 
 #### `$file/{fileId}/init` message
 
-Initialize the file transfer. Server is expected to store metadata from the payload in the session along with `{fileId}` as a reference for the rest of file metadata.
+Initialize the file transfer. Server is expected to store metadata from the payload in the session along with `{fileId}` as a reference for the rest of file transfer.
```

Comment:
May be we should make it more clear that `clientId` + `fileId` acts as a primary key for file transfers on the cluster level, and `fileId` is like a primary key for transfers from the same client.